_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: GSoC 2021 Project
+author: Freyam Mehta
+theme: twitter
+---
+
+{% include JB/setup %}
+
+This blog covers the work done by me during the Google Summer of Code 2021. Dask took part in the program under the NumFOCUS umbrella organization.
+
+Google Summer of Code is a global program focused on bringing more student developers into open source software development. Students work with an open source organization on a 10 week programming project during their break from school.
+
+## Contents
+
+- [Visualizing the Performance Characteristics of Computations](#visualizing-the-performance-characteristics)
+- [Graphical Representation of Task Graphs](#graphical-representation)
+- [HTML Representation of Task Graphs](#html-representation)
+
+## Visualizing the Performance Characteristics of Computations
+
+## Graphical Representation of Task Graphs
+
+I worked on enhancing the graphviz output of the task graphs. The Graphviz library boasts a great set of attributes which can be modifified to create a more visually appealing output. I worked on making them more illustrative, engaging, and informative.
+
+### Fixing calling `.visualize()` with `filename=None` [#7740](https://github.com/dask/dask/pull/7740)
+
+I fixed a minor bug that was caused when users tried to call `dask.visualize()` with `filename=None`. I fixed it by adding an extra condition before it reaches the error line to make sure the format is not `None` by making it equal to the default `png`.
+
+```python
+import dask
+import dask.array as da
+
+array = da.arange(10)
+dask.visualize(array, filename=None)
+```
+
+<img src="/images/gsoc21/7740.png" alt="#7740 Demo" height=414 width=736>
+
+### Add node size scaling to the Graphviz output for the high level graphs [#7869](https://github.com/dask/dask/pull/7869)
+
+I tweaked the node sizes in the High-level graph's task graphs to align with the density of each layer. Layers with more tasks would appear larger than the rest.
+
+```python
+import dask.array as da
+
+array = da.random.random((10000, 10000), chunks=(200, 200))
+result = array + array.T - array.mean(axis=0)
+
+result.dask.visualize()
+```
+
+<img src="/images/gsoc21/7869.png" alt="#7869 Demo" height=414 width=736>
+
+### Change graphviz font family to sans [#7931](https://github.com/dask/dask/pull/7931)
```

Comment:
This change is too minor to include in our update of new features, let's leave it out.